src/main/java/duke/commands/CommandExecution.java

```diff
@@ -0,0 +1,199 @@
+package duke.commands;
+
+import static duke.parser.Parser.dateParser;
+import static duke.parser.Parser.dateTimeParser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import duke.exception.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.tasks.ToDo;
+import duke.ui.Ui;
+
+
+/**
+ * A class representing command execution.
+ */
+public class CommandExecution {
+
+    /**
+     * Executes the commands from the input.
+     *
+     * @param enumCommand the enumeration type of the input command.
+     * @param instruction the string of input command.
+     * @param tasks       the TaskList storing the list of tasks.
+     * @return a string representing the command output.
+     * @throws DukeException if the description or the datetime format of the task is illegal.
+     */
+    public static String executeCommand(EnumCommand enumCommand, String instruction, TaskList tasks)
+            throws DukeException {
+        Ui ui = new Ui();
+        switch (enumCommand) {
+        case TODO:
+            if (instruction.substring(4).strip().equals("")) {
+                throw new DukeException("The description of a todo cannot be empty.");
+            }
+            tasks.add(new ToDo(instruction.substring(4).strip()));
+            return ui.addTaskAlert(tasks);
+
+        case DEADLINE:
+            if (instruction.substring(8).strip().equals("")) {
+                throw new DukeException("The description of a deadline cannot be empty.");
+            }
+
+            String[] tempDeadline = instruction.substring(8).strip().split("/by");
+
+            if (tempDeadline.length < 2) {
+                throw new DukeException("The date and time of the deadline cannot be empty.");
+            }
+
+            String deadlineDescription = tempDeadline[0].strip();
+            String deadlineDateTime = tempDeadline[1].strip();
+            LocalDateTime deadlineDate = dateTimeParser(deadlineDateTime);
+            tasks.add(new Deadline(deadlineDescription, deadlineDate));
+            return ui.addTaskAlert(tasks);
+
+        case EVENT:
+            if (instruction.substring(5).strip().equals("")) {
+                throw new DukeException("The description of an event cannot be empty.");
+            }
+
+            String[] tempEvent = instruction.substring(5).strip().split("/at");
```

Comment:
the naming of a collection should be in the plural form, in this case, tempEvents